### DISCUSSION TITLE
Poster reveal manual problem

### DISCUSSION THREAD

**Discussion by phovorka** [google/model-viewer#3938] [Q&A]
Descr:
I need a poster in my viewer where several models can be shown...but it wont work .....i made example  from: Example ->WebXR Demo together + Example->Customize Load ==> see demo

Live demo:
[https://webxr-demo-model.glitch.me/]

This demo works only when i delete "reveal="manual" ..... but i need to have poster there ....... so how could i do it?

Thanks Emmet

**Comment by elalish** [comment]
You're calling `dismissPoster()` on a div, rather than the model-viewer element.

**Comment by phovorka** [comment]
i forgot add id="lazy-load" to model viewer tag.... but even with this , it still doesnt work properly.... switchSrc problem?

**Comment by phovorka** [comment]
pls any idea?

**Reply by elalish** [reply]
I can't debug your code for you. If you can make a minimal repro of something that should work but doesn't and either post your code or a repro link with specific steps of what you expect and what happens instead, I can probably help. Before you do that, you should fill your code with `console.log` statements and ensure everything you see output makes sense to you.

**Reply by phovorka** [reply]
sorry .... i dont want to use you as a debugger.... 

My Expectation:  this code should change a poster after click on a button.  (Is your WEB XR example  .. )  
- poster will not change   

```
  const modelViewer = document.querySelector("model-viewer");

      window.switchSrc = (element, name) => {
        const base = "https://modelviewer.dev/assets/ShopifyModels/" + name;
        modelViewer.src = base + ".glb";
        modelViewer.poster = base + '.webp';
        const slides = document.querySelectorAll(".slide");
        slides.forEach((element) => {
          element.classList.remove("selected");
        });
        element.classList.add("selected");
      };

      document
        .querySelector(".slider")
        .addEventListener("beforexrselect", (ev) => {
          // Keep slider interactions from affecting the XR scene.
          ev.preventDefault();
        });

      document
        .querySelector("#button-load")
        .addEventListener("click", () =>
          document.querySelector("#lazy-load").dismissPoster()
        );  
```

**Reply by elalish** [reply]
Okay, so when you trigger `switchSrc` you get the same poster, but when you click it the new model appears? That could be a bug; can you provide a link to this code in a live URL so I can verify?

**Reply by phovorka** [reply]
[https://glitch.com/edit/#!/webxr-demo-model](https://glitch.com/edit/#!/webxr-demo-model) just take a look....

i tried another way but the same result...
Changing poster doesnt work  ##3995
